docs/notation.md

```diff
@@ -0,0 +1,79 @@
+# Container Image Sign and Verify with notation tool
+
+| :zap: Requirement | nerdctl >= 1.3.0 |
+|-------------------|------------------|
+
+[notation](https://github.com/notaryproject/notation) is a project to add signatures as standard items in the registry ecosystem, and to build a set of simple tooling for signing and verifying these signatures.
+
+You can enable container signing and verifying features with `push` and `pull` commands of `nerdctl` by using `notation`
+under the hood with make use of flags `--sign` while pushing the container image, and `--verify` while pulling the
+container image.
+
+* Ensure notation executable in your `$PATH`.
+* You can install notation by following this page: https://notaryproject.dev/docs/installation/cli/
+* Notation follows the RC of OCI spec v1.1.0. The container registries rarely have this compliance so far. Follow the [instruction](https://notaryproject.dev/docs/quickstart/#create-an-oci-compatible-registry) to set up the local registry with the compliance for testing purpose.
```

Comment:
Does this mean notation won't work on most public registries currently other than testing on local registry? Do they have any issue tracking registry support? That might help track when we can remove this statement.

Reply:
Actually notation can work in some registries (afaik ECR and ACR) but just without explicit support. Different registry has their own status and tracking issue so it may be not ideal to put lots of issues in comment.
How about I just remove "The container registries rarely have this compliance so far." so we don't have to remove the statement later?

Reply:
SGTM, I just thought notation might have a doc or summary on which registries they support. If they have, it's good to put a link that redirects users to such a notation doc/user. If they don't have, it's fine (probably a user using notation might already know its support/compatibility; nerdctl just streamline the sign/verify process)

Reply:
Yeah I don't find notation having this doc. My understanding is instead of notation supporting registries, notation supports OCI spec. Registries also support OCI spec. They are tracking their progress respectively.